activesupport/lib/active_support/cache.rb

```diff
@@ -504,7 +504,10 @@ def fetch_multi(*names)
 
           writes  = {}
           ordered = names.index_with do |name|
-            reads.fetch(name) { writes[name] = yield(name) }
+            reads.fetch(name) do
+              block_result = yield(name)
+              writes[name] = block_result unless block_result.nil? && options[:skip_nil]
+            end
           end
```

Comment:
I think the same behavior can be achieved with `compact`, yes?

```suggestion
            reads.fetch(name) { writes[name] = yield(name) }
          end
          writes.compact! if options[:skip_nil]
```

Reply:
Yes, that would work too. I'm fine with either way.

Reply:
Ok, let's use `compact!` then. :+1:

Reply:
Done. I pushed up a bad change and when I realized it, I fixed and pushed up the correct change. I think the tests may have run the bad change though.